### PR TITLE
Bugfixes and clean up

### PR DESCRIPTION
**Pull request by Awzaw**
Please make sure your pull request complies with these guidelines:
- * [ ] Use same formatting
- * [ ] Changes must have been tested on PMMP.
- * [ ] Unless it is a minor code modification, you must use an IDE.
- * [ ] Have a detailed title, like "Fix CustomEnchants::getName() must be..."

#### **What does the PR change?**

Fixes for:

```
[22:03:51] [Server thread/CRITICAL]: pocketmine\plugin\PluginException: "pocketmine\event\entity\ProjectileHitBlockEvent does not have a valid handler list" (EXCEPTION) in "src/pocketmine/plugin/PluginManager" at line 772
[22:03:51] [Server thread/DEBUG]: #0 src/pocketmine/plugin/PluginManager(746): pocketmine\plugin\PluginManager->registerEvent(string pocketmine\event\entity\ProjectileHitBlockEvent, PiggyCustomEnchants\EventListener object, integer 1, pocketmine\plugin\MethodEventExecutor object, PiggyCustomEnchants\Main object, boolean 1)
[22:03:51] [Server thread/DEBUG]: #1 PiggyCustomEnchantsSRC/src/PiggyCustomEnchants/Main(283): pocketmine\plugin\PluginManager->registerEvents(PiggyCustomEnchants\EventListener object, PiggyCustomEnchants\Main object)
[22:03:51] [Server thread/DEBUG]: #2 src/pocketmine/plugin/PluginBase(90): PiggyCustomEnchants\Main->onEnable()

```
and

`ErrorException: "Object of class PiggyCustomEnchants\CustomEnchants\CustomEnchants could not be converted to string" (EXCEPTION) in "PiggyCustomEnchantsSRC/src/PiggyCustomEnchants/Main" at line 463`

and

`ErrorException: "Object of class PiggyCustomEnchants\CustomEnchants\CustomEnchants could not be converted to string" (EXCEPTION) in "PiggyCustomEnchantsSRC/src/PiggyCustomEnchants/Main" at line 727`

Fixed using isset() on results of expressions

#### **Testing Environment**

- PHP: 7.2.0RC5
- PMMP: 73e09392b6ad60d49e28bbefd9ec660cd3546297
- OS: macOS

### CODE REVIEW

File: src/PiggyCustomEnchants/EventListener.php [hunk 1]
```diff
@@ -350,12 +351,12 @@ public function onSneak(PlayerToggleSneakEvent $event)
     }
 
     /**
-     * @param ProjectileHitBlockEvent $event
+     * @param ProjectileLaunchEvent $event
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/EventListener.php]: Why? We want this to run when a projectile hits a block and not when it is launched. This is for the grappling enchant.

File: src/PiggyCustomEnchants/EventListener.php [hunk 2]
```diff
@@ -646,7 +647,8 @@ public function checkToolEnchants(Player $player, Event $event)
                 $chance = 10 * $enchantment->getLevel();
                 $random = mt_rand(0, 100);
                 if ($random <= $chance) {
-                    if (isset(self::ORE_TIER[$block->getId()])) {
+					$ore = self::ORE_TIER[$block->getId()];
+                    if (isset($ore)) {
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/EventListener.php]: Look at line 652; the variable is already defined, you can just move it.

File: src/PiggyCustomEnchants/EventListener.php [hunk 3]
```diff
@@ -670,7 +672,8 @@ public function checkToolEnchants(Player $player, Event $event)
                 $finaldrop = array();
                 $otherdrops = array();
                 foreach ($drops as $drop) {
-                    if (isset(self::SMELTED_ITEM[$drop->getId()])) {
+                	$smelted = self::SMELTED_ITEM[$drop->getId()];
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/EventListener.php]: If we're defining this variable, why not use it on line 677.

File: src/PiggyCustomEnchants/Main.php [hunk 4]
```diff
@@ -460,9 +460,10 @@ public function addEnchantment(Item $item, $enchants, $levels, $check = true, Co
                 $levels[] = 1;
             }
         }
-        $combined = array_combine($enchants, $levels);
+        $i = 0;
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/Main.php]: Why???

File: .idea/PiggyCustomEnchants.iml [hunk 5]
```diff
@@ -1,18 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DaPigGuy) [inline .idea/PiggyCustomEnchants.iml]: Why delete the IDE files ;-;

File: src/PiggyCustomEnchants/EventListener.php (hunk 1 above)
Comment (Awzaw) [reply src/PiggyCustomEnchants/EventListener.php]: maybe a PMMP bug, see the description: ProjectileHitBlockEvent does not have a valid handler list

File: src/PiggyCustomEnchants/EventListener.php (hunk 3 above)
Comment (Awzaw) [reply src/PiggyCustomEnchants/EventListener.php]: was in a hurry, did the same for $tier

File: .idea/PiggyCustomEnchants.iml (hunk 5 above)
Comment (Awzaw) [reply .idea/PiggyCustomEnchants.iml]: problem with the gitignore?

File: src/PiggyCustomEnchants/Main.php (hunk 4 above)
Comment (Awzaw) [reply src/PiggyCustomEnchants/Main.php]: `Object of class PiggyCustomEnchants\CustomEnchants\CustomEnchants could not be converted to string`

File: .idea/PiggyCustomEnchants.iml (hunk 5 above)
Comment (Awzaw) [reply .idea/PiggyCustomEnchants.iml]: I didn't even touch the .idea folder, it just vanished!

File: src/PiggyCustomEnchants/EventListener.php (hunk 1 above)
Comment (DaPigGuy) [reply src/PiggyCustomEnchants/EventListener.php]: and ProjectHitEvent is an abstract class 0.0

File: src/PiggyCustomEnchants/Main.php (hunk 4 above)
Comment (DaPigGuy) [reply src/PiggyCustomEnchants/Main.php]: That should not be causing the issue...

**Comment by Aericio** [comment]
@DaPigGuy approve it?

**Comment by DaPigGuy** [comment]
Not yet.

**Comment by Aericio** [comment]
👌

**Pull request closed without merging**